### PR TITLE
API: add parameters perpage and page for api api/org/users

### LINKED ISSUE #32540: API: Support listing org users with proper pagination support
**What would you like to be added**:
Support listing org users with proper pagination support in the HTTP API.

**Why is this needed**:
Having a lot of org users can freeze the browser when navigating to the Organization Users (org/users) in the UI.

**Additional info**:
For `/api/users` a new endpoint was added to not introduce a breaking change, see https://grafana.com/docs/grafana/latest/http_api/user/#search-users-with-paging. Suggest adding a new search with paging endpoint for org users as well.

### PR DESCRIPTION
**Pull request by ying-jeanne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #32540

**Special notes for your reviewer**:
To make it backward compatible, now we still do response.JSON(200, result.Users) to make the response not changed. When frontend change is made, we need to change response.JSON(200, result.Users) to response.JSON(200, result) instead.

### CODE REVIEW

**Comment by ying-jeanne** [comment]
test result
tempsnip

**Comment by ying-jeanne** [comment]
**Note: To make it backward compatible, now we still do response.JSON(200, result.Users) to make the response not changed. When frontend change is made, we need to change response.JSON(200, result.Users) to response.JSON(200, result) instead.**

File: pkg/api/org_users.go [hunk 1]
```diff
@@ -124,33 +113,60 @@ func isOrgAdminFolderAdminOrTeamAdmin(c *models.ReqContext) (bool, error) {
 	return isAdminOfTeamsQuery.Result, nil
 }
 
+func (hs *HTTPServer) GetUsers(c *models.ReqContext) (models.SearchOrgUserQueryResult, error) {
```
Comment (aknuds1) [inline pkg/api/org_users.go:116]: Nit, but I think it's better to simplify the return signature of the method to return the type required by the clients ([]*models.OrgUserDTO). It reduces complexity and the potential for mistakes.
```suggestion
func (hs *HTTPServer) GetUsers(c *models.ReqContext) ([]*models.OrgUserDTO, error) {
```

File: pkg/api/org_users.go [hunk 2]
```diff
@@ -124,33 +113,60 @@ func isOrgAdminFolderAdminOrTeamAdmin(c *models.ReqContext) (bool, error) {
...
+		Limit: perPage,
+		Page:  page,
+	}
+	var err error
+	query.Result.Users, err = hs.getOrgUsersHelper(query, c.SignedInUser)
+
+	query.Result.Page = page
+	query.Result.PerPage = perPage
+
+	return query.Result, err
```
Comment (aknuds1) [inline pkg/api/org_users.go:144]: Nothing is used from `query.Result` but the `Users` field, so best just return that.

```suggestion
	return hs.getOrgUsersHelper(query, c.SignedInUser)
```

File: pkg/models/org_user.go [hunk 3]
```diff
@@ -107,11 +107,18 @@ type UpdateOrgUserCommand struct {
...
 
 type GetOrgUsersQuery struct {
-	OrgId int64
-	Query string
-	Limit int
+	OrgId  int64
+	Query  string
+	Limit  int
+	Page   int
+	Result SearchOrgUserQueryResult
```
Comment (aknuds1) [inline pkg/models/org_user.go:114]: Can we remove the new result type? As you can see from my comments above, you only need to return the users.

File: pkg/services/sqlstore/org_test.go [hunk 4]
```diff
@@ -317,6 +317,38 @@ func TestAccountDataAccess(t *testing.T) {
...
 					})
 				})
+
+			})
+		})
+
+		Convey("Given multiple saved users", func() {
+			setting.AutoAssignOrg = true
+			setting.AutoAssignOrgId = 1
+			setting.AutoAssignOrgRole = "Viewer"
```
Comment (aknuds1) [inline pkg/services/sqlstore/org_test.go:327]: You need to clean up after changing global variables, to not bleed into other tests:
```suggestion
			origAutoAssignOrg = setting.AutoAssignOrg
			origAutoAssignOrgId := setting.AutoAssignOrgId
			origAutoAssignOrgRole := setting.AutoAssignOrgRole
			t.Cleanup(func() {
				setting.AutoAssignOrg = origAutoAssignOrg
				setting.AutoAssignOrgId = origAutoAssignOrgId
				setting.AutoAssignOrgRole = origAutoAssignOrgRole
			})
			setting.AutoAssignOrg = true
			setting.AutoAssignOrgId = 1
			setting.AutoAssignOrgRole = "Viewer"
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Generally looks good AFAICT, but it looks as if some things are made more complicated than they need to be.

**Comment by marefr** [comment]
@ying-jeanne 

> To make it backward compatible, now we still do response.JSON(200, result.Users) to make the response not changed. When frontend change is made, we need to change response.JSON(200, result.Users) to response.JSON(200, result) instead.

That's why I suggested to create a new endpoint in the issue and I still believe we should do that now:
> For /api/users a new endpoint was added to not introduce a breaking change, see https://grafana.com/docs/grafana/latest/http_api/user/#search-users-with-paging. Suggest adding a new search with paging endpoint for org users as well.

**Pull request closed without merging**